File: NiukeAB/AB8/AB8.py
class CirQue():
    def __init__(self):
        self.items=[]

    def push(self, x):
        return self.items.append(x)

    def front(self):
        return self.items[0]

    def pop(self):
        return self.items.pop(0)
    def size(self):
        return len(self.items)
meg=input()
megsplit=meg.split(" ")
lgt=int(megsplit[0])
opt=int(megsplit[1])
sq=CirQue()
for i in range(opt):
    opo= input()
    opsplit=opo.split(" ")
    if opsplit[0]=="push":
        if sq.size() >= lgt:
            print("full")
        else:
            sq.push(int(opsplit[1]))
    elif opsplit[0]=="pop":
        if sq.size()==0:
            print("empty")
        else:
            print(sq.pop())
    elif opsplit[0]=="front":
        if sq.size()==0:
            print("empty")
        else:
            print(sq.front())

File: NiukeAB/AB8/AB8.go
package main

import "fmt"

type CirQue struct {
	items []string
}

func (c *CirQue) Push(item string) {
	c.items = append(c.items, item)
}
func (c *CirQue) Pop() string {
	if c.Size() == 0 {
		return "empty"
	}
	item := c.items[0]
	c.items = c.items[1:]
	return item
}
func (c *CirQue) Size() int {
	return len(c.items)
}
func (c *CirQue) Front() string {
	if c.Size() == 0 {
		return "empty"
	}
	return c.items[0]
}
func main() {
	var cq CirQue
	var lgtn int
	var optn int
	fmt.Scan(&lgtn, &optn)
	for i := 0; i < optn; i++ {
		var op string
		fmt.Scan(&op)
		switch op {
		case "push":
			var x string
			fmt.Scan(&x)
			if cq.Size() > lgtn-1 {
				fmt.Println("full")
			} else {
				cq.Push(x)
			}
		case "pop":
			fmt.Println(cq.Pop())
		case "front":
			fmt.Println(cq.Front())
		}
	}
}
